### PR TITLE
fix: pro-guarding removal of enums 

### PR DESCRIPTION
**Pull request by Shahroz16**
https://github.com/square/moshi#r8--proguard

### CODE REVIEW

File: sdk/src/main/java/io/customer/sdk/data/request/Metric.kt [hunk 1]
```diff
@@ -3,6 +3,7 @@ package io.customer.sdk.data.request
 import com.squareup.moshi.Json
 import com.squareup.moshi.JsonClass
 
+@JsonClass(generateAdapter = false)
```
Comment (levibostian) [inline sdk/src/main/java/io/customer/sdk/data/request/Metric.kt:6]: 🤔 I have only used `generateAdapter = true`. Why are we using `false` here?

Comment (Shahroz16) [reply sdk/src/main/java/io/customer/sdk/data/request/Metric.kt:6]: R8/Pro-guard removes them, hence this is [recommended](https://github.com/square/moshi#r8--proguard) by Moshi. 

Moshi don't need adapters for Enums, it caters them internally.

Comment (levibostian) [reply sdk/src/main/java/io/customer/sdk/data/request/Metric.kt:6]: Ah, 🤦‍♂️. I saw your link (thank you) in the description but didn't see the section titled "Enum". Got it.

**Review by levibostian — APPROVED** [review]
I just confirmed that [we are QA testing r8 compiled builds of the sample app](https://github.com/customerio/RemoteHabits-Android/blob/f396fb57a7cb39b86bb6ec37b713a99a51c4aaaa/fastlane/Fastfile#L103) so we can confirm we are testing our SDK with proguard enabled. Good to double check. With that in mind, LGTM.

**Pull request merged**